File: CONTESTtemplate.py
#!/usr/bin/env python3
# ========================= IMPORTS =========================
import sys
import math
from collections import defaultdict, deque, Counter
from itertools import combinations, permutations, accumulate
from bisect import bisect_left, bisect_right
from heapq import heappush, heappop, heapify
from functools import lru_cache, reduce
from typing import List, Tuple, Set, Dict

# ========================= FAST I/O =========================
input = sys.stdin.readline

def read_int():
    return int(input())

def read_ints():
    return map(int, input().split())

def read_int_list():
    return list(map(int, input().split()))

def read_str():
    return input().strip()

def read_str_list():
    return input().strip().split()

# ========================= CONSTANTS =========================
MOD = 10**9 + 7
MOD2 = 998244353
INF = float('inf')
NINF = float('-inf')
PI = math.pi

# ========================= UTILITY FUNCTIONS =========================

def gcd(a, b):
    """Greatest Common Divisor"""
    while b:
        a, b = b, a % b
    return a

def lcm(a, b):
    """Least Common Multiple"""
    return a * b // gcd(a, b)

def power(a, b, m=MOD):
    """Fast exponentiation: (a^b) % m"""
    res = 1
    a %= m
    while b > 0:
        if b & 1:
            res = (res * a) % m
        a = (a * a) % m
        b >>= 1
    return res

def mod_inv(a, m=MOD):
    """Modular multiplicative inverse using Fermat's little theorem"""
    return power(a, m - 2, m)

def is_prime(n):
    """Check if n is prime"""
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    for i in range(3, int(math.sqrt(n)) + 1, 2):
        if n % i == 0:
            return False
    return True

def sieve(n):
    """Sieve of Eratosthenes - returns list of primes up to n"""
    is_prime = [True] * (n + 1)
    is_prime[0] = is_prime[1] = False
    for i in range(2, int(math.sqrt(n)) + 1):
        if is_prime[i]:
            for j in range(i * i, n + 1, i):
                is_prime[j] = False
    return [i for i in range(n + 1) if is_prime[i]]

def prime_factors(n):
    """Get prime factorization of n"""
    factors = []
    d = 2
    while d * d <= n:
        while n % d == 0:
            factors.append(d)
            n //= d
        d += 1
    if n > 1:
        factors.append(n)
    return factors

def binary_search(arr, target):
    """Binary search - returns index or -1"""
    left, right = 0, len(arr) - 1
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return -1

# ========================= DEBUG =========================
DEBUG = False

def debug(*args):
    """Print debug information"""
    if DEBUG:
        print("[DEBUG]", *args, file=sys.stderr)

# ========================= SOLUTION =========================

def solve():
    """Main solution function"""
    # Your code here
    pass

# ========================= MAIN =========================

def main():
    """Main function"""
    test = 1
    test = read_int()
    
    for tc in range(1, test + 1):
        # print(f"Case #{tc}:", end=" ")
        solve()

if __name__ == "__main__":
    main()

"""
========================= NOTES =========================
- Python is slower than C++, so optimize when needed
- Use PyPy for faster execution (especially for loops)
- Be careful with integer division: // (floor division) vs / (true division)
- For large inputs, use sys.stdin.readline instead of input()
- For recursion, increase limit: sys.setrecursionlimit(10**6)
- For floating point: use math.isclose() for comparisons

========================= COMMON PATTERNS =========================
1. Prefix/Suffix Arrays: list(accumulate(arr))
2. Two Pointers: while left < right
3. Sliding Window: deque for optimal performance
4. Binary Search: bisect_left, bisect_right
5. DFS/BFS: use deque for BFS
6. Dynamic Programming: @lru_cache for memoization
7. Counter for frequency counting
8. defaultdict for default values

========================= TIME COMPLEXITIES =========================
Same as C++, but Python is ~10x slower:
- O(n log n): ~10^5 operations
- O(n): ~10^7 operations
- O(n^2): ~10^3 operations
Use PyPy for better performance!

========================= QUICK SNIPPETS =========================

# Read n integers into list
n = read_int()
arr = read_int_list()

# Read 2D matrix
matrix = []
for _ in range(n):
    row = read_int_list()
    matrix.append(row)

# Prefix Sum
pref = [0] * (n + 1)
for i in range(1, n + 1):
    pref[i] = pref[i-1] + arr[i-1]
# Or use: pref = list(accumulate([0] + arr))

# Prefix XOR
pref = [0] * (n + 1)
for i in range(1, n + 1):
    pref[i] = pref[i-1] ^ arr[i-1]

# Frequency Counter
freq = Counter(arr)
# freq[x] gives count of x

# Default Dict
from collections import defaultdict
d = defaultdict(int)  # default value 0
d = defaultdict(list)  # default value []

# Binary Search on answer
left, right = 0, 10**9
ans = -1
while left <= right:
    mid = (left + right) // 2
    if check(mid):
        ans = mid
        right = mid - 1
    else:
        left = mid + 1

# DFS (recursion)
def dfs(node, visited):
    visited.add(node)
    for neighbor in graph[node]:
        if neighbor not in visited:
            dfs(neighbor, visited)

# BFS
from collections import deque
def bfs(start):
    queue = deque([start])
    visited = {start}
    while queue:
        node = queue.popleft()
        for neighbor in graph[node]:
            if neighbor not in visited:
                visited.add(neighbor)
                queue.append(neighbor)

# Dynamic Programming with memoization
from functools import lru_cache
@lru_cache(maxsize=None)
def dp(n):
    if n <= 1:
        return 1
    return dp(n-1) + dp(n-2)

# Sort with custom key
arr.sort(key=lambda x: x[0])  # sort by first element
arr.sort(key=lambda x: (x[0], -x[1]))  # sort by first asc, second desc

# Matrix rotation (90° clockwise)
matrix = list(zip(*matrix[::-1]))

# Get all divisors of n
def get_divisors(n):
    divisors = []
    for i in range(1, int(math.sqrt(n)) + 1):
        if n % i == 0:
            divisors.append(i)
            if i != n // i:
                divisors.append(n // i)
    return sorted(divisors)

========================= END =========================
"""

File: CONTESTtemplate.cpp
#include <bits/stdc++.h>
using namespace std;

// ========================= MACROS =========================
#define ll long long
#define ull unsigned long long
#define ld long double
#define pb push_back
#define ppb pop_back
#define pf push_front
#define ppf pop_front
#define all(x) (x).begin(), (x).end()
#define rall(x) (x).rbegin(), (x).rend()
#define sz(x) (int)((x).size())
#define endl '\n'

// ========================= LOOPS =========================
#define forn(i, n) for (int i = 0; i < (n); ++i)
#define for1(i, n) for (int i = 1; i <= (n); ++i)
#define rep(i, a, b) for (int i = (a); i < (b); ++i)
#define per(i, a, b) for (int i = (a); i >= (b); --i)
#define trav(a, x) for (auto& a : x)

// ========================= PAIRS =========================
#define F first
#define S second
#define mp make_pair
#define pii pair<int, int>
#define pll pair<ll, ll>
#define pld pair<ld, ld>

// ========================= VECTORS =========================
#define vi vector<int>
#define vll vector<ll>
#define vld vector<ld>
#define vvi vector<vi>
#define vvll vector<vll>
#define vpii vector<pii>
#define vpll vector<pll>

// ========================= MATH =========================
#define mod 1000000007
#define mod2 998244353
#define inf 1e18
#define PI 3.141592653589793238462

// ========================= UTILITY FUNCTIONS =========================
template<typename T> void read(vector<T>& v) { for(auto& x : v) cin >> x; }
template<typename T> void print(const vector<T>& v) { for(const auto& x : v) cout << x << " "; cout << endl; }
template<typename T> T gcd(T a, T b) { return b ? gcd(b, a % b) : a; }
template<typename T> T lcm(T a, T b) { return a / gcd(a, b) * b; }
template<typename T> T power(T a, T b, T m = mod) { T res = 1; while(b) { if(b & 1) res = (res * a) % m; a = (a * a) % m; b >>= 1; } return res; }
ll modInv(ll a, ll m = mod) { return power(a, m - 2, m); }

// ========================= DEBUG =========================
#ifndef ONLINE_JUDGE
#define debug(x) cerr << #x << " = " << (x) << endl;
#define debug2(x, y) cerr << #x << " = " << (x) << ", " << #y << " = " << (y) << endl;
#define debugv(v) cerr << #v << " = "; for(auto& x : v) cerr << x << " "; cerr << endl;
#else
#define debug(x)
#define debug2(x, y)
#define debugv(v)
#endif

// ========================= SOLUTION =========================

void solve() {
    // Your code here
    
}

// ========================= MAIN =========================

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    
    #ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    freopen("error.txt", "w", stderr);
    #endif
    
    int test = 1;
    cin >> test;
    
    for(int tc = 1; tc <= test; tc++) {
        // cout << "Case #" << tc << ": ";
        solve();
    }
    
    return 0;
}

/*
========================= NOTES =========================
- Always check for integer overflow (use ll when needed)
- Be careful with edge cases (n=0, n=1, empty array, etc.)
- Think about time/space complexity before coding
- Test with sample inputs before submitting
- Common bugs: off-by-one errors, integer division, modulo with negative numbers
- For interactive problems: cout.flush() or cout << endl (not '\n')

========================= COMMON PATTERNS =========================
1. Prefix/Suffix Arrays
2. Two Pointers
3. Sliding Window
4. Binary Search
5. DFS/BFS
6. Dynamic Programming
7. Greedy
8. Number Theory (GCD, LCM, Prime Factorization)
9. Segment Trees / Fenwick Trees
10. DSU (Disjoint Set Union)

========================= TIME COMPLEXITIES =========================
O(n log n) - ~10^6 operations: sorting, binary search
O(n) - ~10^8 operations: linear scan
O(n^2) - ~10^4 operations: nested loops
O(n^3) - ~500 operations: 3 nested loops
O(2^n) - ~20 elements: backtracking, subset generation
O(n!) - ~10 elements: permutations

========================= QUICK SNIPPETS =========================

// Binary Search
int l = 0, r = n - 1, ans = -1;
while(l <= r) {
    int mid = l + (r - l) / 2;
    if(check(mid)) {
        ans = mid;
        r = mid - 1;
    } else {
        l = mid + 1;
    }
}

// Prefix Sum
vll pref(n + 1, 0);
for1(i, n) {
    pref[i] = pref[i-1] + arr[i];
}
// sum[l, r] = pref[r] - pref[l-1]

// Prefix XOR
vll pref(n + 1, 0);
for1(i, n) {
    pref[i] = pref[i-1] ^ arr[i];
}
// xor[l, r] = pref[r] ^ pref[l-1]

// GCD of array
int g = arr[0];
for1(i, n-1) g = gcd(g, arr[i]);

// Prime Check
bool isPrime(ll n) {
    if(n < 2) return false;
    for(ll i = 2; i * i <= n; i++)
        if(n % i == 0) return false;
    return true;
}

// Sieve of Eratosthenes
vector<bool> sieve(int n) {
    vector<bool> is_prime(n+1, true);
    is_prime[0] = is_prime[1] = false;
    for(int i = 2; i * i <= n; i++) {
        if(is_prime[i]) {
            for(int j = i*i; j <= n; j += i)
                is_prime[j] = false;
        }
    }
    return is_prime;
}

========================= END =========================
*/
